desktop/core/src/desktop/auth/api_authentications.py

```diff
@@ -125,6 +127,29 @@ def _handle_public_key(self, access_token):
 
             return public_key_pem
 
+  def _handle_jku_ha(self):
+    res = None
+
+    key_server_urls = AUTH.JWT.KEY_SERVER_URL.get()
+    if not key_server_urls:
+      return None
+
+    if "," in key_server_urls:
+      key_server_urls_list = key_server_urls.split(',')
+
+      for jku in key_server_urls_list:
+        try:
+          res = requests.get(jku.rstrip('/'), verify=False)
+        except Exception as e:
+          if 'Failed to establish a new connection' in str(e):
+            LOG.warning('JKU %s is not available.' % jku)
+
+        # Check response for None and if response code is successful (200) or authentication needed (401), use that host URL.
+        if (res is not None) and (res.status_code in (200, 401)):
+          return jku
+    else:
+      # For non-HA, it's normal url string.
+      return key_server_urls
```

Comment:
We should also log an error if none of the url's are available before returning None

Reply:
This block will handle it
https://github.com/cloudera/hue/pull/3360/files#diff-327c16309b48eff8abfa6527b080e3b6aa8260eeccbd3dd3ebb7910f22cae27bR56-R60